kotlinx.coroutines.internal.OnUndeliveredElementKt
    public static final kotlin.jvm.functions.Function1 bindCancellationFun(kotlin.jvm.functions.Function1, java.lang.Object, kotlin.coroutines.CoroutineContext)
    {
        kotlin.jvm.functions.Function1 $r0;
        java.lang.Object $r1;
        kotlin.coroutines.CoroutineContext $r2;
        kotlinx.coroutines.internal.OnUndeliveredElementKt$bindCancellationFun$1 $r3;

        $r0 := @parameter0: kotlin.jvm.functions.Function1;

        $r1 := @parameter1: java.lang.Object;

        $r2 := @parameter2: kotlin.coroutines.CoroutineContext;

        staticinvoke <kotlin.jvm.internal.Intrinsics: void checkNotNullParameter(java.lang.Object,java.lang.String)>($r0, "<this>");

        staticinvoke <kotlin.jvm.internal.Intrinsics: void checkNotNullParameter(java.lang.Object,java.lang.String)>($r2, "context");

        $r3 = new kotlinx.coroutines.internal.OnUndeliveredElementKt$bindCancellationFun$1;

        specialinvoke $r3.<kotlinx.coroutines.internal.OnUndeliveredElementKt$bindCancellationFun$1: void <init>(kotlin.jvm.functions.Function1,java.lang.Object,kotlin.coroutines.CoroutineContext)>($r0, $r1, $r2);

        return $r3;
    }

    public static final void callUndeliveredElement(kotlin.jvm.functions.Function1, java.lang.Object, kotlin.coroutines.CoroutineContext)
    {
        java.lang.Object $r0;
        kotlin.coroutines.CoroutineContext $r1;
        kotlinx.coroutines.internal.UndeliveredElementException $r2;
        kotlin.jvm.functions.Function1 $r3;

        $r3 := @parameter0: kotlin.jvm.functions.Function1;

        $r0 := @parameter1: java.lang.Object;

        $r1 := @parameter2: kotlin.coroutines.CoroutineContext;

        staticinvoke <kotlin.jvm.internal.Intrinsics: void checkNotNullParameter(java.lang.Object,java.lang.String)>($r3, "<this>");

        staticinvoke <kotlin.jvm.internal.Intrinsics: void checkNotNullParameter(java.lang.Object,java.lang.String)>($r1, "context");

        $r2 = staticinvoke <kotlinx.coroutines.internal.OnUndeliveredElementKt: kotlinx.coroutines.internal.UndeliveredElementException callUndeliveredElementCatchingException(kotlin.jvm.functions.Function1,java.lang.Object,kotlinx.coroutines.internal.UndeliveredElementException)>($r3, $r0, null);

        if $r2 == null goto label1;

        staticinvoke <kotlinx.coroutines.CoroutineExceptionHandlerKt: void handleCoroutineException(kotlin.coroutines.CoroutineContext,java.lang.Throwable)>($r1, $r2);

     label1:
        return;
    }

    public static final kotlinx.coroutines.internal.UndeliveredElementException callUndeliveredElementCatchingException(kotlin.jvm.functions.Function1, java.lang.Object, kotlinx.coroutines.internal.UndeliveredElementException)
    {
        kotlin.jvm.functions.Function1 $r0;
        java.lang.Object $r1;
        kotlinx.coroutines.internal.UndeliveredElementException $r2;
        java.lang.Throwable $r3, $r4;
        java.lang.StringBuilder $r5;
        java.lang.String $r6;

        $r0 := @parameter0: kotlin.jvm.functions.Function1;

        $r1 := @parameter1: java.lang.Object;

        $r2 := @parameter2: kotlinx.coroutines.internal.UndeliveredElementException;

        staticinvoke <kotlin.jvm.internal.Intrinsics: void checkNotNullParameter(java.lang.Object,java.lang.String)>($r0, "<this>");

     label1:
        interfaceinvoke $r0.<kotlin.jvm.functions.Function1: java.lang.Object invoke(java.lang.Object)>($r1);

     label2:
        return $r2;

     label3:
        $r3 := @caughtexception;

        if $r2 == null goto label4;

        $r4 = virtualinvoke $r2.<kotlinx.coroutines.internal.UndeliveredElementException: java.lang.Throwable getCause()>();

        if $r4 == $r3 goto label4;

        staticinvoke <kotlin.ExceptionsKt: void addSuppressed(java.lang.Throwable,java.lang.Throwable)>($r2, $r3);

        return $r2;

     label4:
        $r2 = new kotlinx.coroutines.internal.UndeliveredElementException;

        $r5 = new java.lang.StringBuilder;

        specialinvoke $r5.<java.lang.StringBuilder: void <init>()>();

        $r5 = virtualinvoke $r5.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.String)>("Exception in undelivered element handler for ");

        $r5 = virtualinvoke $r5.<java.lang.StringBuilder: java.lang.StringBuilder append(java.lang.Object)>($r1);

        $r6 = virtualinvoke $r5.<java.lang.StringBuilder: java.lang.String toString()>();

        specialinvoke $r2.<kotlinx.coroutines.internal.UndeliveredElementException: void <init>(java.lang.String,java.lang.Throwable)>($r6, $r3);

        return $r2;

        catch java.lang.Throwable from label1 to label2 with label3;
    }

    public static kotlinx.coroutines.internal.UndeliveredElementException callUndeliveredElementCatchingException$default(kotlin.jvm.functions.Function1, java.lang.Object, kotlinx.coroutines.internal.UndeliveredElementException, int, java.lang.Object)
    {
        java.lang.Object $r0, $r1;
        kotlinx.coroutines.internal.UndeliveredElementException $r2;
        kotlin.jvm.functions.Function1 $r3;
        int $i0;

        $r3 := @parameter0: kotlin.jvm.functions.Function1;

        $r0 := @parameter1: java.lang.Object;

        $r2 := @parameter2: kotlinx.coroutines.internal.UndeliveredElementException;

        $i0 := @parameter3: int;

        $r1 := @parameter4: java.lang.Object;

        $i0 = $i0 & 2;

        if $i0 == 0 goto label1;

        $r2 = null;

     label1:
        $r2 = staticinvoke <kotlinx.coroutines.internal.OnUndeliveredElementKt: kotlinx.coroutines.internal.UndeliveredElementException callUndeliveredElementCatchingException(kotlin.jvm.functions.Function1,java.lang.Object,kotlinx.coroutines.internal.UndeliveredElementException)>($r3, $r0, $r2);

        return $r2;
    }

